mode/mscgen/mscgen.js

```diff
@@ -72,11 +72,11 @@
   CodeMirror.defineMIME("text/x-msgenny", {name: "mscgen", language: "msgenny"});
 
   function wordRegexpBoundary(pWords) {
-    return new RegExp("\\b(" + pWords.join("|") + ")\\b", "i");
+    return new RegExp("^\\b(?:" + pWords.join("|") + ")\\b", "i");
```

Comment:
I checked and no caller uses the `[1]` result. That's why it's fine to turn the brackets into a non-capturing group.